server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -202,7 +203,7 @@ private boolean mergeRule(RulesDefinition.Rule def, RuleDto dto) {
     if (mergeDescription(def, dto)) {
       changed = true;
     }
-    if (!dto.getSystemTags().containsAll(def.tags())) {
+    if (!difference(dto.getSystemTags(), def.tags()).isEmpty()) {
```

Comment:
I confirm. It works because afterwards there's the `mergeTags` method that makes different tests. I don't understand why there are 2 places to test system tags.